spock-core/src/main/java/org/spockframework/compiler/condition/BaseVerifyMethodTransform.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spockframework.compiler.condition;
+
+import org.codehaus.groovy.ast.ASTNode;
+import org.codehaus.groovy.ast.MethodNode;
+import org.codehaus.groovy.control.SourceUnit;
+import org.codehaus.groovy.transform.ASTTransformation;
+import org.spockframework.compiler.AstNodeCache;
+import org.spockframework.compiler.ErrorReporter;
+import org.spockframework.compiler.IRewriteResources;
+import org.spockframework.compiler.SourceLookup;
+
+abstract class BaseVerifyMethodTransform implements ASTTransformation {
+
+  static final AstNodeCache nodeCache = new AstNodeCache();
+
+  abstract IVerifyMethodRewriter createRewriter(MethodNode methodNode, IRewriteResources resources);
+
+  @Override
+  public void visit(ASTNode[] astNodes, SourceUnit sourceUnit) {
+    ErrorReporter errorReporter = new ErrorReporter(sourceUnit);
+    SourceLookup sourceLookup = new SourceLookup(sourceUnit);
+
+    try {
+      for (ASTNode node : astNodes) {
+        if (!(node instanceof MethodNode)) {
+          continue;
+        }
+
+        processVerificationHelperMethod((MethodNode) node, errorReporter, sourceLookup);
+      }
+    } finally {
+      sourceLookup.close();
```

Comment:
🤔 should we make `SourceLookup` `AutoClosable` so we could use it in try-with?

Reply:
Sounds reasonable to me, changed in [`69da2e3` (#2112)](https://github.com/spockframework/spock/pull/2112/commits/69da2e3900c13d0e6a3cfc497bd9cd825b561c3b)